### PR TITLE
enhance: enhanced ST_Intersection according to the PostGis Tip. solve…

### PR DESCRIPTION
**Pull request by avshRepo**
… #242

## Is this PR related to a proposed Issue?
yes, i implemented the proposal change

## What changes were proposed in this PR?
i changed ST_Intersection according to PostGis tip

## How was this patch tested?
this patch was tested with the previous ST_Intersection test

## Did this PR include necessary documentation updates?
no

### CODE REVIEW

File: sql/src/main/scala/org/apache/spark/sql/geosparksql/expressions/Functions.scala [hunk 1]
```diff
@@ -215,13 +216,27 @@ case class ST_Intersection(inputExpressions: Seq[Expression])
...
+
+    if (isIntersects && isLeftContainsRight) {
+      return new GenericArrayData(rightgeometry)
+    }
+
+    if (isIntersects && isRightContainsLeft) {
+      return new GenericArrayData(leftgeometry)
+    }
+
+    return new GenericArrayData(GeometrySerializer.serialize(leftgeometry.intersection(rightgeometry)))
```
Comment (jiayuasu) [inline sql/src/main/scala/org/apache/spark/sql/geosparksql/expressions/Functions.scala]: Thank you very much for implementing this!

But I think you forgot one case:

If (!isIntersects) return empty geometry

File: sql/src/main/scala/org/apache/spark/sql/geosparksql/expressions/Functions.scala [hunk 2]
```diff
@@ -215,13 +216,27 @@ case class ST_Intersection(inputExpressions: Seq[Expression])
...
+    val leftgeometry = GeometrySerializer.deserialize(inputExpressions(0).eval(inputRow).asInstanceOf[ArrayData])
+    val rightgeometry = GeometrySerializer.deserialize(inputExpressions(1).eval(inputRow).asInstanceOf[ArrayData])
+
+    val isIntersects = leftgeometry.intersects(rightgeometry)
+    val isLeftContainsRight = leftgeometry.contains(rightgeometry)
+    val isRightContainsLeft = rightgeometry.contains(leftgeometry)
 
+
+    if (isIntersects && isLeftContainsRight) {
+      return new GenericArrayData(rightgeometry)
```
Comment (jiayuasu) [inline sql/src/main/scala/org/apache/spark/sql/geosparksql/expressions/Functions.scala]: This probably should be new GenericArrayData(GeometrySerializer.serialize(rightgeometry)).

I think the current unit test doesn't cover this case. Please add some test cases for it.

Thank you very much!

File: sql/src/main/scala/org/apache/spark/sql/geosparksql/expressions/Functions.scala (hunk 1 above)
Comment (jiayuasu) [reply sql/src/main/scala/org/apache/spark/sql/geosparksql/expressions/Functions.scala]: Yes, I think you can do that. Just make sure you add one test case for this scenario.

File: sql/src/main/scala/org/apache/spark/sql/geosparksql/expressions/Functions.scala (hunk 2 above)
Comment (avshRepo) [reply sql/src/main/scala/org/apache/spark/sql/geosparksql/expressions/Functions.scala]: hi, i created a new pull request with the changes you asked:)

**Comment by avshRepo** [comment]
@jiayuasu I added the changes that you asked

**Pull request merged**